presto-parser/src/main/java/com/facebook/presto/sql/parser/AstBuilder.java

```diff
@@ -989,7 +989,13 @@ public Node visitSpecialDateTimeFunction(SqlBaseParser.SpecialDateTimeFunctionCo
     @Override
     public Node visitExtract(SqlBaseParser.ExtractContext context)
     {
-        return new Extract(getLocation(context), (Expression) visit(context.valueExpression()), Extract.Field.valueOf(context.identifier().getText().toUpperCase()));
```

Comment:
The commit subject should say something like "Throw proper exception when parsing invalid field for EXTRACT", and describe the consequence of now doing that (i.e., CLI can't handle it) in the commit message